### PR TITLE
requests added

### PR DESCRIPTION
**Pull request by MkpMarkovRuslan**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,6 +2,19 @@ let allTasks = JSON.parse(localStorage.getItem('tasks')) || [];
 let valueInput = '';
 let input = null;
 
+window.onload = async function init () {
```
Comment (omanei) [inline index.js]: es6 function

File: index.js [hunk 2]
```diff
@@ -2,6 +2,19 @@ let allTasks = JSON.parse(localStorage.getItem('tasks')) || [];
...
+  input = document.getElementById('add-task');
+  input.addEventListener('change', updateValue);
+  const resp = await fetch('http://localhost:8000/allTasks', {
+    method: 'GET'
+  });
+  let result = await resp.json();
```
Comment (omanei) [inline index.js]: why let?

File: index.js [hunk 3]
```diff
@@ -37,15 +59,20 @@ onClickButton = () => {
...
   localStorage.setItem('tasks', JSON.stringify(allTasks));
-}
+};
+
+deleteTask = async (index) => {
+  const resp = await fetch(`http://localhost:8000/deleteTask?id=${allTasks[index].id}`, {
+    method: 'DELETE'
+  });
+
+  let result = await resp.json();
```
Comment (omanei) [inline index.js]: let?

File: index.js [hunk 4]
```diff
@@ -66,28 +93,45 @@ editTask = (index) => {
     const done = document.createElement('img');
     done.src = './src/done.svg';
     done.id = 'svg';
-    done.addEventListener('click', function () {
+    done.addEventListener('click', async function () {
```
Comment (omanei) [inline index.js]: es6 func

File: index.js [hunk 5]
```diff
@@ -66,28 +93,45 @@ editTask = (index) => {
...
+          body: JSON.stringify(
+            {
+              id: `${allTasks[index].id}`,
+              text: tempInput.value,
+              isCheck: false
+            }
+          )
+        });
+
+        let result = await resp.json();
```
Comment (omanei) [inline index.js]: also let

File: index.js [hunk 6]
```diff
@@ -99,16 +143,15 @@ render = () => {
     container.id = `task-${index}`;
     container.className = 'task-container';
 
-
     const checkBox = document.createElement('input');
     checkBox.type = 'checkbox';
     checkBox.checked = item.isCheck;
     checkBox.onchange = function () {
```
Comment (omanei) [inline index.js]: es6 func

File: index.js [hunk 7]
```diff
@@ -117,8 +160,8 @@ render = () => {
     imageEdit.src = './src/edit.svg';
     container.appendChild(imageEdit);
     imageEdit.addEventListener('click', function () {
```
Comment (omanei) [inline index.js]: also es6

File: index.js [hunk 8]
```diff
@@ -93,7 +93,7 @@ editTask = (index) => {
     const done = document.createElement('img');
     done.src = './src/done.svg';
     done.id = 'svg';
-    done.addEventListener('click', async function () {
+    done.addEventListener('click', async () => {
       if (tempInput.value.trim() === '') {
         alert('Задача это идея - выраженная в словах. Просто напиши их.')
```
Comment (ghost) [inline index.js]: ;

**Pull request merged**